components/organisation-unit-tree/src/organisation-unit-tree/default-render-node-label/label/label.js

```diff
@@ -0,0 +1,118 @@
+import propTypes from 'prop-types'
+import React from 'react'
+import { orgUnitPathPropType } from '../../../prop-types.js'
+import { DisabledSelectionLabel } from './disabled-selection-label.js'
+import { IconizedCheckbox } from './iconized-checkbox.js'
+import { LabelContainer } from './label-container.js'
+import { SingleSelectionLabel } from './single-selection-label.js'
+
+const createNewSelected = ({ selected, path, checked, singleSelection }) => {
+    const pathIndex = selected.indexOf(path)
+
+    if (checked && pathIndex !== -1) return selected
+    if (singleSelection && checked) return [path]
+    if (checked) return [...selected, path]
+    if (pathIndex === -1) return selected
+    if (singleSelection) return []
+    if (selected.indexOf(path) === 0) return selected.slice(1)
+
+    const prevSlice = selected.slice(0, pathIndex)
+    const nextSlice = selected.slice(pathIndex + 1)
+    return [...prevSlice, ...nextSlice]
+}
+
+const Label = ({
+    id,
+    path,
+    open,
+    loading,
+    checked,
+    onChange,
+    dataTest,
+    selected,
+    hasChildren,
+    highlighted,
+    displayName,
+    onToggleOpen,
+    disableSelection,
+    singleSelection,
+    hasSelectedDescendants,
+}) => {
+    const onClick = ({ checked }, event) => {
+        const newSelected = createNewSelected({
+            selected,
+            path,
+            checked,
+            singleSelection,
+        })
+
+        onChange(
+            { id, path, checked, displayName, selected: newSelected },
+            event
+        )
+    }
+
+    if (disableSelection) {
+        return (
+            <LabelContainer highlighted={highlighted}>
+                <DisabledSelectionLabel
+                    label={displayName}
+                    loading={loading}
+                    onToggleOpen={onToggleOpen}
+                />
+            </LabelContainer>
+        )
+    }
+
+    if (singleSelection) {
+        return (
+            <LabelContainer highlighted={highlighted}>
+                <SingleSelectionLabel
+                    checked={checked}
+                    label={displayName}
+                    onChange={onClick}
+                    loading={loading}
+                >
+                    {displayName}
+                </SingleSelectionLabel>
+            </LabelContainer>
+        )
+    }
+
+    return (
+        <LabelContainer highlighted={highlighted}>
+            <IconizedCheckbox
+                dataTest={dataTest}
+                checked={checked}
+                name="org-unit"
+                value={id}
+                label={displayName}
+                loading={loading}
+                indeterminate={!checked && hasSelectedDescendants}
+                onChange={onClick}
+                open={open}
+                hasChildren={hasChildren}
+            />
+        </LabelContainer>
+    )
+}
+
+Label.propTypes = {
+    dataTest: propTypes.string.isRequired,
+    displayName: propTypes.any.isRequired,
```

Comment:
```suggestion
    displayName: propTypes.string.isRequired,
```

Reply:
I chose `any` intentionally so that this can be customized (see the story).
We could(/should?!) take a look at the arguments of the `renderNodeLabel` callback, I assume we can do better there..
Right now, in order to manipulate the node label, the node's `displayName` has to be modified. We could introduce an extra argument (`label`?), which defaults to the `displayName` but can be overwritten with `any`thing

Reply:
I'll add a comment for now